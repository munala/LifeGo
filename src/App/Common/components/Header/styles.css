.header-container {
  width:100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  min-height: 64px;
  align-items: center;
  box-shadow: 0 0 3px 3px rgba(0,0,0,0.2);
  font-family: Roboto,RobotoDraft,Helvetica,Arial,sans-serif;
  background-color: #fff;
}

.left {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  padding-left: 24px;
  flex-basis: 60%;
}

.right {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  padding-right: 24px
}

.logo {
  margin-left: 24px;
  display: 'flex';
  height: 32px;
  width: auto;
  cursor: pointer;
}

.menu-icon {
  display: flex;
  color: #676767;
  cursor: pointer;
}

.logout-icon {
  margin-right: 10px;
}

.vr {
  margin-left: 24px;
  margin-right: 24px;
  display: 'flex';
  height: 32px;
  background-color: rgba(0,0,0,0.1);
  width: 0.5px;
}

.vr-hidden {
  margin-right: 24px;
  display: 'flex';
  height: 32px;
  background-color: rgba(0,0,0,0);
  width: 0.5px;
}

.title-container {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.header-title {
  margin-right: 24px;
  display: 'flex';
  color: #676767;
  font-size: 20px;
  text-decoration: none;
  min-width: 70px;
}

.search-input {
  margin-left: 24px;
  display: 'flex';
  color: #676767;
  font-size: 20px;
}
