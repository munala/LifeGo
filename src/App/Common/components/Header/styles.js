export const styles = theme => ({
  searchInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    color: '#676767',
    fontSize: 20,
    backgroundColor: '#E5E5E5',
    height: 48,
    maxWidth: 720,
    borderRadius: 4,
    paddingLeft: 12,
  },
  margin: {
    margin: theme.spacing.unit,
    cursor: 'pointer',
  },
});

export const otherStyles = {
  avatar: {
    marginLeft: 24,
    height: 48,
    width: 48,
    cursor: 'pointer',
  },
  signIn: {
    marginRight: 24,
  },
};
