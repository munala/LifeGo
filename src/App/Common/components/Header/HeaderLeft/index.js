import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Icon from '@material-ui/core/Icon';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';

import SearchDropDown from '../../../../SearchResults/SearchDropDown';

import '../styles.css';

const HeaderLeft = ({
  menuIconClick,
  history,
  avatarUrl,
  loggedIn,
  to,
  title,
  classes,
  onChange,
  users,
  onSelect,
  searchText,
  viewAll,
}) => (
  <div className="left">
    <Icon
      onClick={menuIconClick}
      className="menu-icon"
    >
    menu
    </Icon>

    <div onClick={() => history.push('/')}>
      <img
        alt="logo"
        className="logo"
        src={avatarUrl}
      />
    </div>

    <div className={loggedIn ? 'vr' : 'vr-hidden'} />

    {
        loggedIn &&
        <Link to={to} className="header-title">{title}</Link>
      }

    <div className="search-input-container">
      <TextField
        id="search"
        className={classes.searchInput}
        onChange={onChange}
        placeholder="Search LifeGo"
        onKeyDown={(e) => {
          if (e.keyCode === 13) { viewAll(); }
       }}
        InputProps={{
          disableUnderline: true,
          style: { flex: 1 },
          startAdornment: (
            <InputAdornment position="start">
              <Icon onClick={menuIconClick} className="menu-icon">search</Icon>
            </InputAdornment>
          ),
        }}
      />
      {
        searchText &&
        <SearchDropDown
          users={users}
          onSelect={onSelect}
          searchText={searchText}
          viewAll={viewAll}
        />
      }
    </div>
  </div>
);

HeaderLeft.propTypes = {
  menuIconClick: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
  avatarUrl: PropTypes.string.isRequired,
  loggedIn: PropTypes.bool.isRequired,
  to: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  classes: PropTypes.shape({}).isRequired,
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  searchText: PropTypes.string.isRequired,
  viewAll: PropTypes.func.isRequired,
  users: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    displayName: PropTypes.string.isRequired,
  })).isRequired,
};

export default HeaderLeft;
