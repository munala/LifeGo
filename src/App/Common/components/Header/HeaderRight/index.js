import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import Badge from '@material-ui/core/Badge';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';

import Avatar from '../../Avatar';
import RaisedButton from '../../RaisedButton';
import avatar from '../../../../../assets/images/user.png';
import { otherStyles } from '../styles';
import '../styles.css';

const HeaderRight = ({
  loggedIn,
  counts,
  classes,
  selectPanel,
  pictureUrl,
  anchorEl,
  handleClose,
  handleClick,
  logout,
  history,
}) => {
  if (loggedIn) {
    return (
      <div className="right">
        <div className="right-icons">
          {
          Object.keys(counts).map(key => (
            <Badge
              key={key}
              className={classes.margin}
              badgeContent={counts[key].count}
              color={counts[key].count ? 'error' : 'secondary'}
              onClick={(event) => {
                selectPanel({
                  event,
                  panel: key,
                });
              }}
            >
              <Icon
                className="menu-icon"
              >
                {counts[key].icon}
              </Icon>
            </Badge>
          ))
        }
        </div>

        <Avatar
          src={pictureUrl || avatar}
          onClick={handleClick}
          style={otherStyles.avatar}
        />
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={handleClose}
        >
          <MenuItem onClick={logout}>
            <Icon
              className="menu-icon logout-icon"
            >
              exit_to_app
            </Icon>
            Logout
          </MenuItem>
        </Menu>
      </div>
    );
  }
  return (
    <RaisedButton
      label="sign in"
      style={otherStyles.signIn}
      onClick={() => history.push({
          pathname: '/login',
          state: { login: true },
        })}
    />
  );
};

HeaderRight.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
  counts: PropTypes.shape({}).isRequired,
  classes: PropTypes.shape({}).isRequired,
  selectPanel: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
  handleClick: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  anchorEl: PropTypes.shape({}),
  pictureUrl: PropTypes.string,
};

HeaderRight.defaultProps = {
  anchorEl: null,
  pictureUrl: null,
};

export default HeaderRight;
