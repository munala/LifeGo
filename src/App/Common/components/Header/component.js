import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Icon from '@material-ui/core/Icon';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import Badge from '@material-ui/core/Badge';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';

import Avatar from '../Avatar';
import RaisedButton from '../RaisedButton';
import { otherStyles } from './styles';
import './styles.css';

class Header extends Component {
  state = {
    anchorEl: null,
    pathname: this.props.location.pathname,
  };

  static getDerivedStateFromProps = ({ location: { pathname } }, state) => ({
    ...state,
    pathname,
  })

  handleClick = (event) => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  }

  logout = () => {
    this.setState({ anchorEl: null });
    this.props.actions.logout();
    this.props.history.push({
      pathname: '/login',
      state: { login: true },
    });
  };

  render() {
    const {
      menuIconClick,
      to,
      avatarUrl,
      classes,
      loggedIn,
      counts,
      onChange,
      history,
      profile: { pictureUrl },
    } = this.props;

    const { pathname, anchorEl } = this.state;

    const titles = {
      '/': 'Home',
      '/home': 'Home',
      '/explore': 'Discover',
      '/mylists': 'My Lists',
      '/profile': 'Profile',
      '/settings': 'Settings',
    };

    const title = titles[pathname];

    return (
      <div className="header-container" >
        <div className="left">
          <Icon
            onClick={menuIconClick}
            className="menu-icon"
          >
          menu
          </Icon>

          <div onClick={() => history.push('/')}>
            <img
              alt="logo"
              className="logo"
              src={avatarUrl}
            />
          </div>

          <div className={loggedIn ? 'vr' : 'vr-hidden'} />

          {
            loggedIn &&
            <Link to={to} className="header-title">{title}</Link>
          }

          <TextField
            id="search"
            className={classes.searchInput}
            onChange={onChange}
            placeholder="Search LifeGo"
            InputProps={{
            disableUnderline: true,
            startAdornment: (
              <InputAdornment position="start">
                <Icon onClick={menuIconClick} className="menu-icon">search</Icon>
              </InputAdornment>
            ),
          }}
          />
        </div>
        {
        loggedIn ?
          <div className="right">
            {
              Object.keys(counts).map(key => (
                <Badge
                  key={key}
                  className={classes.margin}
                  badgeContent={counts[key].count}
                  color={counts[key].count ? 'error' : 'secondary'}
                >
                  <Icon
                    onClick={counts[key].onClick}
                    className="menu-icon"
                  >
                    {counts[key].icon}
                  </Icon>
                </Badge>
              ))
            }

            <Avatar
              src={pictureUrl || require('../../../../assets/images/user.png')} // eslint-disable-line global-require
              onClick={this.handleClick}
              style={otherStyles.avatar}
            />
            <Menu
              id="simple-menu"
              anchorEl={anchorEl}
              open={Boolean(anchorEl)}
              onClose={this.handleClose}
            >
              <MenuItem onClick={this.logout}>
                <Icon
                  className="menu-icon logout-icon"
                >
                  exit_to_app
                </Icon>
                Logout
              </MenuItem>
            </Menu>
          </div> :
          <RaisedButton
            label="sign in"
            style={otherStyles.signIn}
            onClick={() => history.push({
              pathname: '/login',
              state: { login: true },
            })}
          />
      }
      </div>
    );
  }
}

Header.propTypes = {
  menuIconClick: PropTypes.func.isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  profile: PropTypes.shape({
    pictureUrl: PropTypes.string,
  }).isRequired,
  actions: PropTypes.shape({
    logout: PropTypes.func.isRequired,
  }).isRequired,
  to: PropTypes.string.isRequired,
  avatarUrl: PropTypes.string.isRequired,
  classes: PropTypes.shape({}).isRequired,
  loggedIn: PropTypes.bool.isRequired,
  counts: PropTypes.shape({}).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Header;
