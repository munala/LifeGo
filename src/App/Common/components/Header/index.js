import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import Badge from '@material-ui/core/Badge';
import { withStyles } from '@material-ui/core/styles';

import Avatar from '../Avatar';
import RaisedButton from '../RaisedButton';
import './styles.css';

const styles = theme => ({
  searchInput: {
    display: 'flex',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    color: '#676767',
    fontSize: 20,
    backgroundColor: '#E5E5E5',
    height: 48,
    maxWidth: 720,
    borderRadius: 4,
    paddingLeft: 12,
  },
  margin: {
    margin: theme.spacing.unit * 2,
    cursor: 'pointer',
  },
});

const otherStyles = {
  avatar: {
    marginLeft: 24,
    height: 48,
    width: 48,
    cursor: 'pointer',
  },
  signIn: {
    marginRight: 24,
  },
};

const Header = ({
  menuIconClick,
  title,
  avatarUrl,
  classes,
  loggedIn,
  counts,
  onChange,
}) => (
  <div className="header-container" >
    <div className="left">
      <Icon
        onClick={menuIconClick}
        className="menu-icon"
      >
        menu
      </Icon>

      <img
        alt="logo"
        className="logo"
        src={avatarUrl}
      />

      <div className="vr" />

      <div className="header-title">{title}</div>

      <TextField
        id="search"
        className={classes.searchInput}
        onChange={onChange}
        placeholder="Search LifeGo"
        InputProps={{
          disableUnderline: true,
          startAdornment: (
            <InputAdornment position="start">
              <Icon onClick={menuIconClick} className="menu-icon">search</Icon>
            </InputAdornment>
          ),
        }}
      />
    </div>
    {
      loggedIn ?
        <div className="right">
          {
            Object.keys(counts).map(key => (
              <Badge
                key={key}
                className={classes.margin}
                badgeContent={counts[key].count}
                color={counts[key].count ? 'error' : 'secondary'}
              >
                <Icon
                  onClick={counts[key].onClick}
                  className="menu-icon"
                >
                  {counts[key].icon}
                </Icon>
              </Badge>
            ))
          }

          <Avatar
            src={require('../../../../assets/images/user.png')} // eslint-disable-line global-require
            onClick={() => {}}
            style={otherStyles.avatar}
          />
        </div> :
        <RaisedButton
          label="sign in"
          style={otherStyles.signIn}
        />
    }
  </div>
);

Header.propTypes = {
  menuIconClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  avatarUrl: PropTypes.string.isRequired,
  classes: PropTypes.shape({}).isRequired,
  loggedIn: PropTypes.bool.isRequired,
  counts: PropTypes.shape({}).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default withStyles(styles)(Header);
