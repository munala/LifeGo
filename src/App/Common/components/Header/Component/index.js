import React from 'react';
import Popover from '@material-ui/core/Popover';
import Modal from '@material-ui/core/Modal';

import BaseClass from './BaseClass';
import Chat from '../../Chat/Container';
import Notifications from '../../Notifications/Container';
import UserAlerts from '../../UserAlerts/Container';
import HeaderLeft from '../HeaderLeft';
import HeaderRight from '../HeaderRight';
import propTypes from '../propTypes';
import '../styles.css';

class Header extends BaseClass {
  state = {
    anchorEl: null,
    topAnchor: null,
    pathname: this.props.location.pathname,
    loggedIn: this.props.loggedIn, // eslint-disable-line react/no-unused-state
    profile: {}, // eslint-disable-line react/no-unused-state
    selectedPanel: '',
  };

  static getDerivedStateFromProps = ({
    location: { pathname },
    loggedIn,
    actions: {
      getProfile, getNotifications, getAlerts, getConversations,
    },
    profile,
    currentApiCalls,
  }, state) => {
    if (loggedIn && (loggedIn !== state.loggedIn || (!profile.id && !currentApiCalls))) {
      getProfile();
      getNotifications();
      getAlerts();
      getConversations();
    }

    return ({
      ...state,
      profile,
      pathname,
    });
  }

  componentDidMount = () => {
    const {
      getProfile, getNotifications, getAlerts, getConversations,
    } = this.props.actions;

    getProfile();
    getNotifications();
    getAlerts();
    getConversations();
  }

  renderPanels =() => {
    const { selectedPanel, topAnchor } = this.state;

    if (selectedPanel && selectedPanel !== 'conversations') {
      const panels = {
        notifications: () => <Notifications />,
        alerts: () => <UserAlerts />,
      };
      const Panel = panels[selectedPanel];
      return (
        <Popover
          open={Boolean(topAnchor)}
          anchorEl={topAnchor}
          onClose={() => this.selectPanel({
            event: { target: null },
            panel: selectedPanel,
          })}
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
        >
          <Panel />
        </Popover>
      );
    }
    return false;
  }

  render() {
    const {
      menuIconClick,
      to,
      avatarUrl,
      classes,
      loggedIn,
      counts,
      onChange,
      history,
      profile: { pictureUrl },
    } = this.props;

    const { pathname, anchorEl, selectedPanel } = this.state;

    const titles = {
      '/': 'Home',
      '/home': 'Home',
      '/explore': 'Explore',
      '/mylists': 'My Lists',
      '/profile': 'Profile',
      '/settings': 'Settings',
    };

    const title = titles[pathname] || 'Home';

    return (
      <div className="header-container" >
        <HeaderLeft
          menuIconClick={menuIconClick}
          history={history}
          avatarUrl={avatarUrl}
          loggedIn={loggedIn}
          to={to}
          title={title}
          classes={classes}
          onChange={onChange}
        />

        <HeaderRight
          loggedIn={loggedIn}
          counts={counts}
          classes={classes}
          pictureUrl={pictureUrl}
          anchorEl={anchorEl}
          history={history}
          handleClose={this.handleClose}
          handleClick={this.handleClick}
          logout={this.logout}
          selectPanel={this.selectPanel}
        />

        {this.renderPanels()}

        <Modal
          open={selectedPanel === 'conversations'}
          onClose={() => this.selectPanel({
            event: { target: null },
            panel: selectedPanel,
          })}
        >
          <Chat />
        </Modal>
      </div>
    );
  }
}

Header.propTypes = propTypes;

export default Header;
