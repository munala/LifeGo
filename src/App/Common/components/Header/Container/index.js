import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';

import Header from '../Component';
import { logout } from '../../../../Auth/actions';
import { getProfile } from '../../../../Profile/actions';
import { getNotifications } from '../../../../Notifications/actions';
import { getConversations } from '../../../../Chat/actions';
import { getAlerts } from '../../../../UserAlerts/actions';
import { searchUsers, clearSearch } from '../../../../SearchResults/actions';
import { styles } from '../styles';

const mapStateToProps = ({
  loggedIn,
  profile,
  searchResults: { users },
  currentApiCalls: { profile: currentApiCalls },
  ...state
}) => {
  let count = 0;

  const keys = ['conversations', 'notifications', 'alerts'];

  const counts = {
    alerts: {
      count: 1,
      icon: 'person',
    },
    conversations: {
      count: 2,
      icon: 'message',
    },
    notifications: {
      count: 3,
      icon: 'notifications',
    },
  };

  keys.forEach((key) => {
    if (key === 'conversations') {
      state[key].forEach((conversation) => {
        let unread = false;

        conversation.messages.forEach((message) => {
          if (!message.read && message.receiverId === profile.id) {
            unread = true;
          }
        });

        if (unread) { count += 1; }
      });
    } else {
      state[key].forEach((item) => {
        if (!item.read) {
          count += 1;
        }
      });
    }

    counts[key].count = count;
    count = 0;
  });

  return ({
    loggedIn,
    profile,
    counts,
    users,
    currentApiCalls,
  });
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    logout,
    getProfile,
    getNotifications,
    getConversations,
    getAlerts,
    searchUsers,
    clearSearch,
  }, dispatch),
});

export default withRouter(withStyles(styles)(connect(mapStateToProps, mapDispatchToProps)(Header)));
