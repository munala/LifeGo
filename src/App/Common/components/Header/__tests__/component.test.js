import React from 'react';
import { shallow } from 'enzyme';
import Header from '../container';

describe('Header tests', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Header
      menuIconClick={() => {}}
      title="home"
      avatarUrl="url"
      loggedIn
      counts={{}}
      onChange={() => {}}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
