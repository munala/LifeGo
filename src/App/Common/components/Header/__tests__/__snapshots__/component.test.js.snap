// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders without crashing 1`] = `
<Header
  avatarUrl="url"
  classes={
    Object {
      "margin": "Header-margin-2",
      "searchInput": "Header-searchInput-1",
    }
  }
  counts={Object {}}
  loggedIn={true}
  menuIconClick={[Function]}
  onChange={[Function]}
  title="home"
/>
`;
